### PR TITLE
Zmove

### PR DESCRIPTION
**Pull request by VeteranPadgett**
The first commit is just communicating zmove choices to the server. The rest is to try and make the UI change when the zmove button is pressed. I haven't been able to test the last change yet, but the first two seem to work fine. If someone can test it for me would be great. Otherwise, I'll update this once I'm able to do the test myself.

### CODE REVIEW

**Comment by turbedi** [comment]
I can test the PR today if you are busy (I am at home and able to test in about 6 hours)

**Comment by MidwayMarshall** [comment]
mega grey? Design changes to something that was just done :c

File: src/com/podevs/android/poAndroid/pokeinfo/MoveInfo.java [hunk 1]
```diff
@@ -208,6 +263,22 @@ void fillByte(int i, byte b) {
 		});
 	}
 
+	static boolean zpowerloaded = false;
+	private static void loadPokeZPowers() {
+		if (zpowerloaded) {
+			return;
+		}
+		testLoad();;
```
Comment (turbedi) [inline src/com/podevs/android/poAndroid/pokeinfo/MoveInfo.java]: a semicolon too much

**Comment by turbedi** [comment]
Unfortunately I am not able to test it, because I got a new router and I wasn't able to figure out how to Port Forward on it, so I am not able to host a gen 7 server to test your PR :/

**Comment by VeteranPadgett** [comment]
same lol. I can get Fuzzy to host a server for me again when I see him.

For the mega_grey, I wanted to make the zmove button toggle so that you can look at the move description of the zmove and changed the mega button to do the same in the process.

**Comment by Fuzzysqurl** [comment]
you guys know you can connect via localhost as long as you're on the same net? even without a forwarded port. or is the android client that dumb?

**Comment by MidwayMarshall** [comment]
localhost is a computer's own address. Not the intranet address devices have to a router

**Comment by Fuzzysqurl** [comment]
Hm I don't remember what I did when I did webclient on my phone on my server awhile back then. Nevermind I'm dumb .

**Comment by MidwayMarshall** [comment]
You can manually type in something like 192.168.1.10:5080 w/e the computer's internal address is

**Comment by turbedi** [comment]
Did a quick test. After using a Z-Move the move selection when activating the Z-Move button stays and I am not able to choose any other moves anymore. Screenshot:
f1ea52cf-4a99-4a69-80ac-dd57a33c2a10

**Comment by VeteranPadgett** [comment]
rip. Hopefully this fixes it.

**Comment by turbedi** [comment]
Too tired to test your fix right now. Gonna do it tomorrow

**Comment by VeteranPadgett** [comment]
Finally figured out how to connect to your own server from the virtual android. The emulator uses 10.0.2.2 to communicate with your actual computer.

**Comment by VeteranPadgett** [comment]
Well, I still can't test yet cause my server is acting totally wrong. My local is the same as the remote though and in turbedi's picture it looks fine.

**Comment by turbedi** [comment]
Now nothing happens when I press the Z-Move button :/

**Comment by VeteranPadgett** [comment]
Fixed my server by uninstalling and reinstalling qt creator. Tested out newest changed it seems to work!

**Comment by turbedi** [comment]
Now when a pokemon holds a Z-Crystal, I am only able to select moves that are working with the Z-Crystal, no matter if I activate it or not

**Comment by turbedi** [comment]
Nevermind, I didn't update my server, everything is working fine now (as far as I can tell)

**Comment by VeteranPadgett** [comment]
Which is the remaining comment?

**Comment by turbedi** [comment]
Whoops, I created the comment but forgot to press `Submit review`

File: src/com/podevs/android/poAndroid/pokeinfo/ItemInfo.java [hunk 2]
```diff
@@ -116,6 +116,18 @@ public static short memoryChipForType(int type) {
             return memoryChips[type];
         }
 
+        private static short zMoves[] = new short[] {
+            673, 674, 675, 676, 677, 678, 679, 680, 681, 682, 683, 684, 685, 686, 687,
+                688, 689, 690, 691, 692, 693, 694, 695, 696, 697, 698, 699, 700, 701
+        };
+
+        public static short zCrystalMove(short item) {
+            if (item < 3000 || item > 3028) {
```
Comment (VeteranPadgett) [inline src/com/podevs/android/poAndroid/pokeinfo/ItemInfo.java]: done

**Comment by turbedi** [comment]
@MidwayMarshall Merge?

**Comment by VeteranPadgett** [comment]
Updated to not conflict with hidden power changes.

**Comment by turbedi** [comment]
Could you change the behavior of how Mega-Evolitons/Z-moves are selected to how it was before? I think the way how it is done right now is a bit confusing.
Once this is done I'm gonna merge the PR, so only adding missing translations to prevent crashes and some minor cleanups are left before a new client release.

**Comment by VeteranPadgett** [comment]
Where the button disappears upon click? The biggest problem with the zmoves is what if someone clicks on it to use Z-Normal attack but then changes their mind and wants to use Z-Fire attack instead? Any move that can't become a zmove won't be allowed and they have no way to unselect the zmove option. I changed mega evo button just so the two could work the same. If you really want the mega evo button reverted I'll do it, but I think it would be a bad idea for the zmove one.

**Comment by MidwayMarshall** [comment]
The way I made mega work did very much include a way to revert it. Press menu and uncheck mega

**Comment by VeteranPadgett** [comment]
Oh, I didn't know about that. I'll make them both where you can uncheck in menu then.

**Comment by VeteranPadgett** [comment]
That should do it

**Comment by turbedi** [comment]
Did you test your last commit? If yes, I'm gonna merge it.

After that the client should be ready for a new release 🎉 🎉 🎉 🎉

**Comment by VeteranPadgett** [comment]
Yes, I tested it. If you need to do translations make sure to add this one before release.

<string name="zmove">Z-Move</string>

**Pull request merged**